bpf/bpf_sock.c

```diff
@@ -595,12 +701,10 @@ static __always_inline int __sock6_xlate(struct bpf_sock_addr *ctx)
 
 	svc = __lb6_lookup_service(&key);
 	if (!svc) {
-		__builtin_memset(&key.address, 0, sizeof(key.address));
+		ctx_get_v6_address(ctx, &key.address);
```

Comment:
Do we need to fetch this twice? The `key.address` should remain unchanged after the `__lb6_lookup_service()`.

Reply:
The port is actually reset, see: https://github.com/cilium/cilium/blob/b2c9b075ec92b0ed820c6f31cf358c268e95fff7/bpf/lib/lb.h#L682

But I'll remove the address.

Reply:
Ah, yes, sounds good!